spec/Model.spec.js

```diff
@@ -414,231 +355,456 @@ describe('Model', () => {
+        category.organization = organization
+        expect(category.organization).toEqual(organization)
       })
 
-      const todo2 = store.createOrUpdateModelFromData({
-        type: 'todos',
-        id: 2,
-        attributes: { description: 'Todo 2' },
-        relationships: { notes: { included: false } }
+      it('sets a relationship object via relationships hash', () => {
+        expect(category.organization).toBeUndefined()
+        category.relationships.organization = { data: { id: organization.id, type: 'organizations' } }
+        category.relationships.organization = { data: { id: organization2.id, type: 'organizations' } }
+        expect(category.organization).toEqual(organization2)
+        category.relationships.organization = null
+        expect(category.organization).toBeUndefined()
+        category.relationships.organization = { data: { id: organization.id, type: 'organizations' } }
+        expect(category.organization).toEqual(organization)
       })
 
-      expect(todo1.notes.map(n => n.attributes.description)).toEqual(['Note 1 for Todo 1', 'Note 2 for Todo 1'])
-      expect(todo2.notes.map(n => n.attributes.description)).toEqual(['Note 1 for Todo 2'])
-    })
+      it('keeps a relationship after saving', async () => {
+        expect(category.organization).toBeUndefined()
+        category.relationships.organization = { data: { id: organization.id, type: 'organizations' } }
 
-    it('relationship data is cached when falling back to inverse relationships', () => {
-      // For this test, we need to create seed data using JSON in order to simulate the results of API calls
-      // where Notes are loaded separately, and Organizations don't include notes directly.
-      store.createOrUpdateModelFromData({
-        type: 'notes',
-        id: 10,
-        attributes: { description: 'Note 1 for Todo 100' },
-        relationships: { todo: { data: { type: 'todos', id: 100 } } }
+        mockTodoResponse = JSON.stringify({ data: category.jsonapi({ relationships: ['organization'] }) })
+        fetch.mockResponseOnce(mockTodoResponse)
+        await category.save({ relationships: ['organization'] })
+
+        expect(category.organization).toEqual(organization)
       })
 
-      store.createOrUpdateModelFromData({
-        type: 'notes',
-        id: 11,
-        attributes: { description: 'Note 2 for Todo 100' },
-        relationships: { todo: { data: { type: 'todos', id: 100 } } }
+      it('relationship can be set', () => {
+        note.todo = todo
+        expect(note.todo).toEqual(todo)
       })
 
-      const todo1 = store.createOrUpdateModelFromData({
-        type: 'todos',
-        id: 100,
-        attributes: { description: 'Todo 100' },
-        relationships: { notes: { included: false } }
+      it('relationship can be unset', () => {
+        note.todo = todo
+        expect(note.todo).toEqual(todo)
+
+        note.todo = null
+        expect(note.todo).toBeFalsy()
```

Comment:
Duplicated

Reply:
I'm going to keep it. testing in different contexts.